### ISSUE TITLE
Black color around the bounding box of the map displaying .mbtiles tiles served by Martin

### ISSUE THREAD

**Issue by nouryf** [urbica/martin#922]
Hi,

I installed Martin on a Linux box and created a map using LeafLet to display raster tiles (.jpg) from an .mbtiles file. When the map is displayed I see black color on the left and right of the map. If I change the tile server to use [mbtiles-sever](https://github.com/tobinbradley/mbtiles-server) to serve the tiles, the black color is not displayed.

Martin:
![image](https://github.com/maplibre/martin/assets/18246160/a0c871a7-0407-4b16-8d84-cc199da8a7a4)

mbtiles-sever:
![image](https://github.com/maplibre/martin/assets/18246160/44d3c907-847d-42fe-86cb-489cc0e1708d)

Could this be a onfiguration issue ?

I checked the .mbtiles projection using gdalinfo and it's Pseudo Mercator:
![image](https://github.com/maplibre/martin/assets/18246160/1f89ee04-286e-443e-a704-98854fd44758)

Thanks for any help on this

**Comment by sharkAndshark** [comment]
Hi @nouryf   :)
Would you mind to share your code and MbTiles file?

**Comment by nouryf** [comment]
Thanks @sharkAndshark for your response. I tried to make a smaller version of the .mbtiles but did not duplicate the issue with the result (shape of the bounding box more rectangular in comparison to close to square with the .mbtiles showing the issue).

So I'm sharing the original dataset [here](https://drive.google.com/file/d/1JdQVDqapIJh-ucLHAC9ytaCBci4006Kp/view?usp=sharing). The .zip is 106 mb (114 mb uncompressed). 

I'm using the code from [this project](https://github.com/tobinbradley/timemachine) to display the .mbtiles (served by Martin). The code is using Leaflet as the .js client code.

To duplicate the issue:
- Start Martin serving the shared .mbtiles
- Clone the [client code](https://github.com/tobinbradley/timemachine) and make the following edits:
* src/assets/surveys.js to change the url for the served mbtiles:

```json
[
  {
  "flydate": "2018-08-17",
  "url":"http://localhost:3000/Amz/{z}/{x}/{y}",
  "minzoom":14,
  "maxzoom":19
  },
  {
    "flydate": "2019-08-17",
    "url":"http://localhost:3000/Amz/{z}/{x}/{y}",
    "minzoom":14,
    "maxzoom":19
  }
]
```

* Edit /src/store.js to put the starting map location (line 90):
export let mapLocation = writable([-8.2411,31.20527,14])
* Edit src/lib/Map.svelte to change the min and max zoom (lines 42 to 63):

```js
    if (!leftLayer || leftLayer._url !== l.url) {
      if (leftLayer) leftLayer.remove()
      leftLayer = L.tileLayer(l.url, {
        attribution: l.attribution,
        **maxZoom: 19,**
        maxNativeZoom: l.maxzoom,
        minNativeZoom: l.minzoom,
        **minZoom: 14**
      })
    }
    if (!rightLayer || rightLayer._url !== r.url) {
      if (rightLayer) rightLayer.remove()
      rightLayer = L.tileLayer(r.url, {
        attribution: r.attribution,
        **maxZoom: 19,**
        maxNativeZoom: r.maxzoom,
        minNativeZoom: r.minzoom,
        **minZoom: 14**
      }).addTo(map)
    }
```

- Start the client server: npm run dev
- Browse to the output url and u should black colors on the left and right.

Thanks

**Comment by nyurik** [comment]
Overall, this sounds extremely suspicious.  Martin does not do any kind of tile data manipulation - it simply returns whatever was given in the store. The only things that MAY affect the drawing is some metadata, e.g. if tilejson values get messed up somehow, and they change the output for some reason (???), or if there are some HTTP headers that are not being set correctly for images, and again change the output?  IMO, any mistake should prevent displaying of the tiles alltogether, so this is really confusing.  I hope @sharkAndshark can reproduce it

**Comment by sharkAndshark** [comment]
I duplicated it.
<img width="1446" alt="截屏2023-10-09 下午5 35 43" src="https://github.com/maplibre/martin/assets/42399654/93127b23-9d2a-46be-a6ff-c5311ff97808">

<img width="1429" alt="截屏2023-10-09 下午5 50 30" src="https://github.com/maplibre/martin/assets/42399654/cc5e62f5-3e00-4001-bdd5-45992e69d310">


The black color is within these tiles:
* http://localhost:3000/Amz/15/15631/13391
* http://localhost:3000/Amz/15/15631/13392
* http://localhost:3000/Amz/15/15631/13390
* http://localhost:3000/Amz/15/15631/13393


﻿I doubt the black color is not by martin but in this `MbTiles` file as I can see the black color with [mapbox/mbview](https://github.com/mapbox/mbview) either.   

Still need more proof: I will check the `MbTiles` file directly with navicat and QGIS(downloading) later.

<img width="1429" alt="截屏2023-10-09 下午5 46 09" src="https://github.com/maplibre/martin/assets/42399654/211da044-0deb-457c-8d25-b1bdd9ffc73d">

**Comment by nouryf** [comment]
Thanks @sharkAndshark for looking into this and duplicating it. Using QGIS I don't see those black colors. I see only few slim areas where black colors is present on the edges:

![image](https://github.com/maplibre/martin/assets/18246160/27570f54-01ca-4d87-a6e1-f15ca1577b6d)

**Comment by sharkAndshark** [comment]
@nouryf The default zoom in QGIS (the scale is 1:28883, we can calculate the zoom as it's a 3857 grid set ) when viewing this mbtiles is not equals to the zoom in the js client(zoom 14-19).  I think this is the reason why the black color is thin in QGIS and wid in js client.  
I will run a SQL query to inspect the tile bytes directly.

**Comment by sharkAndshark** [comment]
The images extracted from mbtiles file.
Zoom 12:

![2421](https://github.com/maplibre/martin/assets/42399654/24a30c94-c514-4488-9ad4-cbf442d6dd43)

**Comment by sharkAndshark** [comment]
> @nouryf The default zoom in QGIS (the scale is 1:28883, we can calculate the zoom as it's a 3857 grid set ) when viewing this mbtiles is not equals to the zoom in the js client(zoom 14-19). I think this is the reason why the black color is thin in QGIS and wid in js client. I will run a SQL query to inspect the tile bytes directly.

I'm wondering QGIS has a different rendering strategy?

**Comment by nouryf** [comment]
As far as QGIS goes, probably it's making black color for not fully covered tiles transparent and hence not showing the black colors for those tiles.

@sharkAndshark I now confirmed that the black color is coming from tiles that are not fully covered by the image pixels. I was able to duplicate this with mbtiles-sever after retrying it (I was not able to do so before I was using a .png version of the mbtiles that has alpha band).

So this is not a Martin's bug at all. It's error user (mine).

Thanks again @sharkAndshark for looking into this.
